dvc/exceptions.py

```diff
@@ -323,3 +323,7 @@ def __init__(self, path_info):
                 format_link("https://man.dvc.org/config#cache"),
             )
         )
+
+
+class IsADirectoryError(DvcException):
+    """Raised when a file operation is requested on a directory."""
```

Comment:
This is the same error as [IsADirectoryError](https://docs.python.org/3/library/exceptions.html#IsADirectoryError). But, it does not inherit from `OSError` for obvious reasons. I did not feel good to raise `InvalidArgumentError` because the argument was not invalid in the strict sense, it's property was.

I can remove this if we don't need this though. Too many exceptions, I know. :slightly_smiling_face:

Reply:
@Suor, regarding the comments below, `IsADirectoryError` is an `OSError`, and since this is not a system-related error/IO failure and we are only using it internally, I really didn't feel comfortable to have both `DvcException` and `OSError` in the inheritance chain. I only wanted to maintain the semantics of it.

Reply:
It depends on how you look at it. Anyway, what is the purpose of this PR?

Reply:
> this is not a system-related error/IO failure

As I see by usage it 's I/O error - we cannot read the dir. What is the purpose of creating custom `IsADirectoryError`? I think if we need custom exception it should has another abstraction level depends on the needs

Reply:
> Anyway, what is the purpose of this PR?

To make every exception we throw to be `DvcException`.